# HG changeset patch
# User Rik <rik@octave.org>
# Date 1511831616 28800
#      Mon Nov 27 17:13:36 2017 -0800
# Node ID b366fd9fb3ee7b0975c7a3e1e22a45b171a22765
# Parent  c381bca75c649e59af96a048304e72a85da18c46
meshz.m: Change indices to be one-based, not zero-based, when single input Z given.

* meshz.m: Change docstring to note that for single argument Z,
x = 1:columns (Z), y = 1:rows (Z).  Change code to follow suit.

diff --git a/scripts/plot/draw/meshz.m b/scripts/plot/draw/meshz.m
--- a/scripts/plot/draw/meshz.m
+++ b/scripts/plot/draw/meshz.m
@@ -25,17 +25,17 @@
 ## @deftypefnx {} {@var{h} =} meshz (@dots{})
 ## Plot a 3-D wireframe mesh with a surrounding curtain.
 ##
 ## The wireframe mesh is plotted using rectangles.  The vertices of the
 ## rectangles [@var{x}, @var{y}] are typically the output of @code{meshgrid}.
 ## over a 2-D rectangular region in the x-y plane.  @var{z} determines the
 ## height above the plane of each vertex.  If only a single @var{z} matrix is
 ## given, then it is plotted over the meshgrid
-## @code{@var{x} = 0:(columns (@var{z}) - 1), @var{y} = 0:(rows (@var{z}) - 1}).
+## @code{@var{x} = 1:columns (@var{z}), @var{y} = 1:rows (@var{z})}.
 ## Thus, columns of @var{z} correspond to different @var{x} values and rows
 ## of @var{z} correspond to different @var{y} values.
 ##
 ## The color of the mesh is computed by linearly scaling the @var{z} values
 ## to fit the range of the current colormap.  Use @code{caxis} and/or
 ## change the colormap to control the appearance.
 ##
 ## Optionally the color of the mesh can be specified independently of @var{z}
@@ -66,24 +66,24 @@ function h = meshz (varargin)
 
   if (isempty (charidx))
     charidx = nargin + 1;
   endif
 
   if (nargin == 1)
     z = varargin{1};
     [m, n] = size (z);
-    x = 0:(n-1);
-    y = (0:(m-1)).';
+    x = 1:n;
+    y = (1:m).';
     c = z;
   elseif (nargin == 2)
     z = varargin{1};
     [m, n] = size (z);
-    x = 0:(n-1);
-    y = (0:(m-1)).';
+    x = 1:n;
+    y = (1:m).';
     c = varargin{2};
   elseif (charidx == 4)
     x = varargin{1};
     y = varargin{2};
     z = varargin{3};
     c = z;
   else
     x = varargin{1};
